### PR TITLE
WebUI shows the groups and superusers

### PR DESCRIPTION
**Pull request by bliuredhat**
The users will hide and show in their group botton; the superusers_list
include all superusers.

JIRA: PDC-1762

### CODE REVIEW

File: pdc/apps/auth/templates/api_perms.html [hunk 1]
```diff
@@ -21,14 +28,86 @@
...
           <td>{{ resource }}</td>
           <td><a href="{{ perms.url }}">{{ perms.url }}</a></td>
-          <td>{% for user in perms.read %}{{ user }}<br />{% endfor %} </td>
-          <td>{% for user in perms.create %}{{ user }}<br />{% endfor %}</td>
-          <td>{% for user in perms.update %}{{ user }}<br />{% endfor %}</td>
-          <td>{% for user in perms.delete %}{{ user }}<br />{% endfor %}</td>
+          <td>{% if '@all' in perms.read %}
+                  <em> @all </em> <br />
+              {% else %}
+                {% for group, value in perms.create.items %}
```
Comment (simozhan) [inline pdc/apps/auth/templates/api_perms.html]: perms.create.items > perms.read.items

Comment (simozhan) [inline pdc/apps/auth/templates/api_perms.html]: maybe the perms.read part should be similar with perms.create

File: pdc/apps/auth/views.py [hunk 2]
```diff
@@ -257,7 +264,7 @@ def api_perms(request):
     Render API resource perms.
     """
     context = {
-        'api_perms': get_api_perms(request),
+        'api_perms': get_api_perms(request)
```
Comment (bliuredhat) [inline pdc/apps/auth/views.py]: yes, fixed in 93b5d7d.

File: pdc/apps/auth/views.py [hunk 3]
```diff
@@ -140,22 +140,26 @@ def user_profile(request):
...
+    inactive_user = {str(user.username) for user in models.User.objects.filter(is_active=False)}
+    superusers_set = {str(user.username) for user in models.User.objects.filter(is_superuser=True)}
     try:
         group_resource_permission_list = get_list_or_404(models.GroupResourcePermission,
                                                          resource_permission=resource_permission)
         groups_list = [str(obj.group.name) for obj in group_resource_permission_list]
-        users_set = {user.username for user in get_user_model().objects.filter(groups__name__in=groups_list)}
-        users_list = list(superusers_set.union(users_set) - inactive_user)
+        for obj in groups_list:
+            group_popover.setdefault('@' + obj, _get_users_mapping_group(obj, inactive_user))
```
Comment (simozhan) [inline pdc/apps/auth/views.py]: is this _get_users_mapping_group only used here, do we need to add a method for it?
and it seems the function for the method is to exclude the inactive_user from a group

File: pdc/apps/auth/views.py [hunk 4]
```diff
@@ -236,11 +240,14 @@ def get_api_perms(request):
...
-            members = get_users_and_groups(obj)
-            groups = ["@" + group_name for group_name in members['groups']]
-            members_list = sorted(members['users'] + groups)
-        perms.setdefault(name, OrderedDict()).setdefault(obj.permission.name, set()).update(members_list)
+            members_list = get_users_and_groups(obj)
+        perms.setdefault(name, OrderedDict()).setdefault(obj.permission.name, members_list)
         perms.setdefault(name, OrderedDict()).setdefault('url', url)
+    for perm, value in perms.iteritems():
+        tmp_perm = {'read', 'create', 'update', 'delete', 'url'} - set(value.keys())
+        if list(tmp_perm):
```
Comment (simozhan) [inline pdc/apps/auth/views.py]: it doesn't need to add this if, following for loop will cover this

File: pdc/apps/auth/views.py [hunk 5]
```diff
@@ -140,22 +140,26 @@ def user_profile(request):
...
+        for obj in groups_list:
+            group_popover.setdefault('@' + obj, _get_users_mapping_group(obj, inactive_user))
     except Http404:
-        groups_list = []
-        users_list = list(superusers_set - inactive_user)
-    members['groups'] = groups_list
-    members['users'] = users_list
-    return members
+        pass
+    group_popover.setdefault('superusers_list', sorted(list(superusers_set - inactive_user)))
```
Comment (bliuredhat) [inline pdc/apps/auth/views.py]: fixed in 380881c.

File: pdc/apps/auth/views.py (hunk 4 above)
Comment (bliuredhat) [inline pdc/apps/auth/views.py]: fixed in 
	380881c

File: pdc/apps/auth/templates/api_perms.html [hunk 6]
```diff
@@ -3,6 +3,13 @@
 
 {% block content %}
 <h1>API Permissions</h1>
+<div>
+  <ul>
+    <li><strong>@all</strong> means that all users and groups have the permission here.</li>
+    <li><strong>superusers_list</strong> list all superusers who have all permissions.</li>
```
Comment (simozhan) [inline pdc/apps/auth/templates/api_perms.html]: Is it better to change superusers_list to @superusers

**Comment by simozhan** [comment]
LGTM

**Comment by ycheng-aa** [comment]
Looks good.

**Pull request merged**